File: javascript/meetup/meetup.js
var weekDays = "Sunday Monday Tuesday Wednesday Thursday Friday Saturday".split(" ");

var Meetup = function(year, month, day, nth) {
	/* Finds a meetup date */
  var intDay = weekDays.indexOf(day);

  switch (nth) {
    case "teenth":
      return find_day(year, month, intDay, 13, 19);
    case "1st":
      return find_day(year, month, intDay, 1, 7);
    case "2nd":
      return find_day(year, month, intDay, 8, 14);
    case "3rd":
      return find_day(year, month, intDay, 15, 21);
    case "4th":
      return find_day(year, month, intDay, 22, 28);
    case "5th":
      return find_day(year, month, intDay, 29, 31);
    case "last":
      return last(year, month, intDay);
  }
};


module.exports = Meetup;

File: ruby/meetup/meetup.rb
class Meetup
  def initialize(month, year)
    @cal = (Date.new(year, month, 1)..Date.new(year, month, -1))
  end

  def day(meet_up, schedule)
    all_meet_ups = @cal.select { |date| date.send("#{meet_up}?") }
    case schedule
    when :first  then all_meet_ups[0]
    when :second then all_meet_ups[1]
    when :third  then all_meet_ups[2]
    when :fourth then all_meet_ups[3]
    when :last   then all_meet_ups.last
    # when :teenth
      all_meet_ups.select { |d| d.mday > 12 && d.mday < 20 }.first
    end
  end
end
